### ISSUE TITLE
Keeping Device awake, disable sleep, Samsung Galaxy A11 mod for battery removal

### ISSUE THREAD

**Issue by geniusled** [feelfreelinux/octo4a#462]
Hello All, first of all, thank you so much for offering such a platform to Octoprint on android. i think its a great use for a extra phone or old phone.

context: 
i have acquired some Samsung Galaxy A11 with defective battery (spicy pillows, puffed up battery). i was able to modify them to run without a battery (TLDR: removed lithium cell, supplying 4V at battery terminal through BMS, can use built in USB C port to power device).   
i have tried many configurations types in order to utilise the USB C for both Power and OTG simultaneously ( as many people here). i have come to conclusion that this device cannot use the "dock mode" of OTG (needed for charging while offering OTG). i can confirm the model is at cause because similar setups used on a galaxy S8 work fine. 
Galaxy A11 specs:  
- Supports OTG (tested)
- supports fast charging (tested)
- can run Octo4a (tested on battery enabled device, connects fine to printer)
- can run android 10-11 or 12
- 32gb built in memory, 2gb ram

Issue:
since i cannot charge and otg simultaneously,
i can use a separate power supply to power the phone through the battery terminals(the phone is not charging, cannot use the dev mode stay awake while charging). But because Android tries to do battery optimisation, this causes Octo4a to only run for a few days before the app is killed by os.

Question:
How can i keep phone "awake" with display turned on with octo4A app when the device is not considered "charging"? within the screen options, only up to 30 minutes is available before sleep. when connected to charger via USB port, with dev option stay awake, the phone will stay awake for weeks with octo4A app running.

things i will investigate in parrallel to your comments:
- ADB shell via wifi
- - command to keep display alive
- - command to fake device charging state < likely to kill otg
- - others
- Tasker type apps to force keep alive

**Comment by geniusled** [comment]
# discoveries so far:
 TLDR: can keep device from sleeping (until next boot) using dev mode options and adb shell.
**WARNING: doing this will disable battery charging functionality until next boot**

> ### start with:
> atleast 10% battery or more
> non rooted device 
> not plugged in on charger 
> display timeout at 15seconds for example) 

### on the android device:
developper mode:
- enable the "stay awake" feature
- enable the wireless debugging feature (Android 11 feature) (can also be done USB but instructions here are wireless)
- inside wireless debug option, pair with code

### on the computer via android SDK(ADB)
via ADB cmd line (on computer)
- download and use adb via command line
- do adb pairing with > adb pair "ipaddress:port"
- do adb connect with > adb connect "ipaddress:port"
- open adb shell with > adb -s "ipaddress:port" shell
- in adb shell type >  dumpsys battery set -f ac 1 (fakes android device being on AC charger)
- close debugging connection (optional)

### Result:
(these are not reboot persistant)
- phone does not display charging in notifications bar
- (if low battery, low battery notifications dissapear)
- phone display now stays on all the time    < what i am looking for!
- otg connection still possible in this state
- side effect: battery charging disabled (not an issue for me considering external power)

### preliminary conclusion:
more investigation necessary. 
Potentially can combine the usage of "stay awake" dev mode setting with adb shell command to fake AC charger presence.
 (question: can a tasker automate these things to eliminate need for adb shell on pc?)

**Comment by geniusled** [comment]
so far device continously works (for context: samsung A11 ,cut out battery, feeding 4v into battery contacts(bms), printer connected via otg)

since the fake ac charger presence only works until reboot of device, i built this batch script to make it easier to do this when the device gets rebooted.
note: the "wireless debugging" option and adb pairing on android also disables itself after boot.
note2: this assumes that you have downloaded the ADB standalone package [here](https://developer.android.com/tools/releases/platform-tools) and have it unzipped in the same level as the script

```
@echo off
echo:
echo this script will enable stay awake, fake AC charger presence and switch back to previous app on Android phone using ADB commands
echo:

echo workflow:
echo step 1: go to settings / developper options/
echo step 2: enable wifi debugging
echo step 3: tap on wifi debugging
echo step 4: tap on pairing with code
pause
echo:

REM Getting the info from user
set /p ip=Enter phone IP address: 
echo: 
set /p connectport=Enter phone connection port: 
echo: 
echo Now press on pairing with code:
set /p code=Enter pairing code: 
echo: 
set /p port=Enter phone pairing port: 
echo: 

REM Doing the ADB commands.........
echo Pairing with device....
.\platform-tools_r34.0.5-windows\platform-tools\adb.exe pair %ip%:%port% %code%
echo:
echo Connecting to device....
.\platform-tools_r34.0.5-windows\platform-tools\adb.exe connect %ip%:%connectport%
echo:
echo Sending Commands to stay awake, fake AC charger, switch back to previous app
.\platform-tools_r34.0.5-windows\platform-tools\adb.exe -s %ip%:%connectport% shell "settings put global stay_on_while_plugged_in 3 && dumpsys battery set -f ac 1 && input keyevent KEYCODE_APP_SWITCH && input keyevent KEYCODE_APP_SWITCH"
echo:
echo Closing connection.....
.\platform-tools_r34.0.5-windows\platform-tools\adb.exe disconnect %ip%:%connectport%
echo:
echo Completed
```

question: is there a way for these commands to be executed from within Octoprint app? or can this be automated to run from within the device on boot? it would save from needing to run through this process every time

**Comment by geniusled** [comment]
Im going to close this issue as the previous solution i mentioned (android adb , set charger to AC) has been up and working for a month now.
the only downside is that it needs to be done at every boot as i have yet to implement a solution that is automated on boot of the phone without rooting it.
